### DISCUSSION TITLE
x and y axis labels

### DISCUSSION THREAD

**Discussion by jsa91** [peterhinch/micropython-nano-gui#60] [Q&A]
Hi,

I'm using class `TSequence` and think it is great. I'm wondering if there is a smart way to add integer labels to my axis? As of now, I'm adding my labels in a loop but i believe it is quite ineffective:

```python
x_axis_division = [15, 30, 48, 66, 83, 98, 115, 133, 150, 168, 186, 204]
y_axis_division = [254, 230, 205, 185, 160]

for x_label, x_axis_division in enumerate(x_axis_division):
    Label(self.wri, 285, x_axis_division, str((x_label * 2)), align=2)
    
for y_label, y_axis_division in enumerate(y_axis_division):
    Label(self.wri, y_axis_division, 4, str((y_label)), align=2)
```

**Comment by peterhinch** [comment]
I can't see any obvious way to improve this. When you say it's "ineffective" what do you mean?

**Reply by jsa91** [reply]
Thank you for your answer. I thought, since I instantiate a new `Label()` for every iteration my memory load gets larger than it has to be.

**Reply by peterhinch** [reply]
The `Label` class is quite lightweight, it is a wrapper for the `Writer` or `CWriter` class. You could use the latter directly, but the savings would be small. In my own applications I use `Label` instances.
